frontend/src/metabase/query_builder/actions/querying.js

```diff
@@ -109,7 +109,7 @@ export const runQuestionQuery = ({
 
     const cardIsDirty = originalQuestion
       ? question.isDirtyComparedToWithoutParameters(originalQuestion) ||
-        question.card().id == null
+        question.id() == null
```

Comment:
This looks promising.
https://github.com/metabase/metabase/blob/92badc0138e40e32275a82b6d16d9305329928b4/frontend/src/metabase-lib/Question.ts#L900-L902

There are literally around 10 places that we operate on question's `.card()` properties directly. I don't think that's intended.